workflow/artifacts/s3/s3.go

```diff
@@ -77,6 +90,16 @@ func (s3Driver *S3ArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string
 func (s3Driver *S3ArtifactDriver) Save(path string, outputArtifact *wfv1.Artifact) error {
 	err := wait.ExponentialBackoff(wait.Backoff{Duration: time.Second * 2, Factor: 2.0, Steps: 5, Jitter: 0.1},
 		func() (bool, error) {
+			var err error
+			var bucket []byte
+			if outputArtifact.S3.BucketSecret.Key != "" {
```

Comment:
*Nit/Suggestion* Maybe we can make a helpful utility func to `getBucket` (as this logic is repeated above https://github.com/argoproj/argo/pull/1692/files#diff-0d7017944420a424dbd879fc0ef1c0ddR44)

Reply:
:+1: on @lorenanicole's comment, might be worth abstracting this to another function where we can do proper validation/creation/fetching.